Metalama.Framework.Engine/CodeModel/Builders/NamedTypeBuilder.cs

```diff
@@ -246,7 +251,14 @@ public override SyntaxTree PrimarySyntaxTree
                         List<UsingDirectiveSyntax>(),
                         List<AttributeListSyntax>(),
                         List<MemberDeclarationSyntax>() ),
-                    path: this.FullName + ".cs",
+                    path: this.TypeParameters.Count > 0
+                        ? $"{this.FullName}`{this.TypeParameters.Count}.cs"
+                        : $"{this.FullName}.cs",
+                    options: this.Compilation.RoslynCompilation.SyntaxTrees.FirstOrDefault() switch
+                    {
+                        { Options.Features: { } features } => CSharpParseOptions.Default.WithFeatures( features ),
```

Comment:
I think you should just use the options from the syntax tree, since other properties have to match as well, like language version.

```suggestion
                        { Options: var options } => options,
```